source/SH3/system/glcontext.cpp

```diff
@@ -17,11 +17,109 @@
 #include "SH3/system/glcontext.hpp"
 #include "SH3/system/log.hpp"
 #include "SH3/system/window.hpp"
+#include "SH3/glcallback.hpp"
 #include <GL/glew.h>
 #include <GL/glu.h>
 #include <GL/glext.h>
 #include <glm/gtc/matrix_transform.hpp>
 
+/**
+ *  OpenGL error callback function. Used in conjunction with glDebugMessageCallback
+ *
+ *  @arg source     The Source of this error
+ *  @arg type       The type of error this is
+ *  @arg id         Error ID
+ *  @arg severity   How severe this error is. If it's fatal, terminate the program with @ref die()
+ *  @arg length     Length of the message log
+ *  @arg message    Error message passed to us from OpenGL
+ *  @arg data       User Param data
+ */
+__glCallback debugCallback(GLenum source, GLenum type, GLuint id, GLenum severity, GLsizei length, const GLchar *message, const void *data)
```

Comment:
This can be static. And while you probably intended to have this function as `extern "C"`, the `APIENTRY`-macro is probably what we want to use here.

Reply:
> the APIENTRY-macro is probably what we want to use here.

The few tutorials I've seen have used this as well (though on my system it's `GLAPIENTRY`). Would you prefer having a custom `extern "C"` type we can use across files or are you okay with functions being defined as `static extern "C" void GLAPIENTRY` (I did this mostly for readability)

Reply:
> And while you probably intended to have this function as extern "C", the APIENTRY-macro is probably what we want to use here.

`APIENTRY` (or `GLAPIENTRY`) expands to __stdcall on my system. Is there a particular purpose to this (or why it's necessary for this OpenGL function??)

Reply:
It seems I was looking at a tutorial for C and Windows, because we need both `extern "C"` and `GL_APIENTRY`. On my system `APIENTRY` and `GL_APIENTRY` are both defined to be the same thing, though the name `GL_APIENTRY` gives more context anyways.
`__stdcall` specifies a certain way to pass arguments and the return value.